File: cpp/cpp_1000-/1561.cc
// C++
// leetcode 1561
// https://ssg.leetcode.cn/problems/maximum-number-of-coins-you-can-get/

class Solution {
public:
    int maxCoins(vector<int>& piles) {
    	sort(piles.begin(), piles.end());
    	int n = piles.size(), res = 0, chances = n / 3;
    	do {
    		res += piles[n - 2 * chances];
    	} while (--chances);
    	return res;
    }
};

File: Go/Go_1000-/1561.go
import "sort"

// Go
// leetcode 1561
// https://ssg.leetcode.cn/problems/maximum-number-of-coins-you-can-get/

func maxCoins(piles []int) int {
	sort.Ints(piles)
	n := len(piles)
	turn, res, p := n/3, 0, n-2
	for i := 0; i < turn; i++ {
		res += piles[p]
		p -= 2
	}
	return res
}